ad-joining/register-computer/join.ps1

```diff
@@ -27,13 +27,61 @@
 #--------------------------------------------------------------------------------------------
 
 $ErrorActionPreference = "Stop"
+$InformationPreference = "Continue";
+
+$metadataUri = "http://metadata.google.internal/computeMetadata/v1/instance";
+$metadataHeaders = @{"Metadata-Flavor" = "Google"};
+
+$enableDiagnostics = $false;
+try
+{
+    $enableDiagnostics = (Invoke-RestMethod -Headers $metadataHeaders `
+        -Uri "$($metadataUri)/attributes/enable-adjoin-diagnostics");
+
+    if($enableDiagnostics)
+    {
+        Write-Information -MessageData "AD Join diagnostics: Enabled"; 
+
+        $diagnosticsBucket = (Invoke-RestMethod -Headers $metadataHeaders `
+            -Uri "$($metadataUri)/attributes/adjoin-diagnostics-bucket");
+
+        if($diagnosticsBucket -eq "")
```

Comment:
Instead of checking for both metadata keys, could we just use `adjoin-diagnostics-bucket`, and enable diagnostics if it's set? That might simplify the code a bit.

Reply:
As mentioned in my PR comment I also like this better. Will change.